File: 02-11-2024_Kth_distance.cpp
#include <unordered_map>
#include <vector>

class Solution {
public:
    bool checkDuplicatesWithinK(const std::vector<int>& arr, int k) {
        std::unordered_map<int, int> dict;

        for (int i = 0; i < arr.size(); ++i) {
            if (dict.find(arr[i]) != dict.end() && i - dict[arr[i]] <= k) {
                return true;
            }
            dict[arr[i]] = i;
        }

        return false;
    }
};

File: 02-11-2024_Kth_distance.py
class Solution:
    def checkDuplicatesWithinK(self, arr, k):
        dict={}
        for i in range(0,len(arr)):
            if (arr[i] in dict and i-dict[arr[i]]<=k):
                return 1
            dict[arr[i]]=i
        return 0
